stages/org.osbuild.bootupd

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/python3
+"""
+Install GRUB on both BIOS and UEFI systems,
+ensuring that your bootloader stays up-to-date.
+
+Bootupd supports updating GRUB and shim for
+UEFI firmware on x86_64 and aarch64,
+and GRUB for BIOS firmware on x86_64.
+The project is deployed in Fedora CoreOS and derivatives
+"""
+
+import os
+import subprocess
+import sys
+
+import osbuild.api
+from osbuild.util import ostree
+
+SCHEMA_2 = r"""
+"devices": {
+  "type": "object",
+  "additionalProperties": true
+},
+"mounts": {
+  "type": "array"
+},
+"options": {
+  "additionalProperties": true,
+  "properties": {
+    "deployment": {
+      "additionalProperties": false,
+      "required": ["osname", "ref"],
+      "properties": {
+        "osname": {
+          "description": "Name of the stateroot to be used in the deployment",
+          "type": "string"
+        },
+        "ref": {
+          "description": "OStree ref to create and use for deployment",
+          "type": "string"
+        },
+        "serial": {
+          "description": "The deployment serial (usually '0')",
+          "type": "number",
+          "default": 0
+        }
+      }
+    },
+    "static-configs": {
+      "description": "Install the grub configs defined for Fedora CoreOS",
+      "type": "boolean"
+    },
+    "bios": {
+      "additionalProperties": false,
+      "required": ["disk"],
+      "properties": {
+        "disk": {
```

Comment:
NITPICK: Since this option maps to a "device" attached to the stage, perhaps we should call it `device` instead of `disk`?  The description could also be updated to mention that it needs to match the name of a device:
```
Name of stage device to install GRUB for BIOS-based systems.
```
Alternatively, we can force the device name in the schema for the `devices` and omit this (like we do in the `mkfs` stages).  Unless we expect that we might ever have multiple devices.

Reply:
new upload